src/tuf.rs

```diff
@@ -292,32 +291,32 @@ impl Tuf {
         match hash_alg {
             HashType::Sha512 => {
                 Self::read_and_verify(&mut file,
-                                      &mut Sha512::new(),
+                                      Context::new(&SHA512),
                                       target_meta.length,
                                       &expected_hash.0)
             }
             HashType::Sha256 => {
                 Self::read_and_verify(&mut file,
-                                      &mut Sha256::new(),
+                                      Context::new(&SHA256),
                                       target_meta.length,
                                       &expected_hash.0)
             }
             HashType::Unsupported(_) => Err(Error::NoSupportedHashAlgorithms),
         }
     }
 
-    fn read_and_verify<R: Read, D: Digest>(input: &mut R,
-                                           digest: &mut D,
-                                           size: i64,
-                                           expected_hash: &[u8])
-                                           -> Result<(), Error> {
+    fn read_and_verify<R: Read>(input: &mut R,
+                                mut context: Context,
```

Comment:
IMO it would be clearer to qualify it as `digest::Context` because there are (will be) several `Context` types.